src/main/java/io/stargate/sgv2/jsonapi/service/processor/HybridFieldExpander.java

```diff
@@ -42,24 +43,51 @@ public static void expandHybridField(CommandContext context, Command command) {
   }
 
   // protected for testing purposes
-  protected static void expandHybridField(int docIndex, int docCount, JsonNode docNode) {
+  protected static void expandHybridField(
+      CommandContext context, int docIndex, int docCount, JsonNode docNode) {
     final JsonNode hybridField;
 
-    if ((docNode instanceof ObjectNode doc)
-        && (hybridField = doc.remove(DocumentConstants.Fields.HYBRID_FIELD)) != null) {
-      switch (hybridField) {
-        case NullNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
-        case TextNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
-        case ObjectNode ob -> addFromObject(doc, ob, docIndex, docCount);
-        default ->
-            throw ErrorCodeV1.HYBRID_FIELD_UNSUPPORTED_VALUE_TYPE.toApiException(
-                "expected String, Object or `null` but received %s (Document %d of %d)",
-                JsonUtil.nodeTypeAsString(hybridField), docIndex + 1, docCount);
+    if (docNode instanceof ObjectNode doc) {
+      // Check for $hybrid field
+      if ((hybridField = doc.remove(DocumentConstants.Fields.HYBRID_FIELD)) != null) {
+        switch (hybridField) {
+            // this is {"$hybrid" : null}
+          case NullNode ignored -> addLexicalAndVectorize(doc, hybridField, hybridField);
+          case TextNode ignored -> {
+            context.commandFeatures().addFeature(CommandFeature.HYBRID);
```

Comment:
we can add the feature if the if statement is true, which means the user include $hybrid in the command.